doc/tutorials/ostro-oic-tutorial/step4/tutorial.md

```diff
@@ -0,0 +1,371 @@
+ports an initial value when application starts. This is usually useful,
+but for our application it would be annoying. We want to retrieve our
+initial value from server, not send an initial value to server. Thus,
+we disable this behaviour by setting this option to `false`.
+Time to connect buttons to accumulators:
+
+```sh
+increase_button OUT -> IN _(boolean/filter) TRUE -> INC power_value
+decrease_button OUT -> IN _(boolean/filter) TRUE -> DEC power_value
+```
+
+As buttons outputs `true` packets when pressed and `false` when released,
+we need to filter its output: only when they are pressed matters.
+We achieve this with node type `boolean/filter` - connect only its
+TRUE output port to get only true packets.
+Ports `INC` and `DEC` increases and decreases accumulator current value,
+respectively.
+Print some debug information when change value:
+
+```sh
+power_value OUT -> IN _(console:prefix="Power value changed: ")
+```
+
+## Receiving command line arguments and connecting to OIC server
+
+It is time to connect to OIC server. Node type `oic/client-light` will
+help is with such task. But it needs to know where connect. Different
+from HTTP, it doesn't need a network address, but an ID, something
+on the form '2f3089a1dbfb43d38cab64383bdf9380'. When running our server,
+it informs us its ID:
+
+```sh
+$ sol-fbp-runner server.fbp oic-and-http-light-server.fbp
+OIC device-id: 0bacbaedb34b434785dec5519d60101c (string)
+note: http-light-server running on port 8002, paths: /state and /power (string)
+```
+
+But how to inform it to our client controller? We could hardcode this
+information for tests, but on real world, this is unpractical - even for
+tests. Soletta inform us about command line arguments, so let's use them:
+
+```sh
+cmdline_arguments(app/argc-argv)
+_(constant/int:value=1) OUT -> IN cmdline_arguments
+```
+
+Node type `app/argc-argv` is the one we use to get this information.
+Second line states that we want first argument. Now, we can properly
+use `oic/client-light` node type:
+
+```sh
+oic(oic/client-light)
+cmdline_arguments OUT -> DEVICE_ID oic
+```
+
+First line declares our OIC client and second one sets the DEVICE_ID
+it should connect from command line argument we received.
+Now, cross-link accumulator and OIC client nodes:
+
+```sh
+oic OUT_POWER -> SET power_value
+power_value OUT -> IN_POWER oic
+```
+
+To complete our controller first iteration, let's inform user
+when server value changes:
+
+```sh
+oic OUT_POWER -> IN _(console:prefix="Server power level is: ")
+```
+
+This way, if another controller changes server 'power' value, we
+are informed. That's a nice OIC feature: we can observe the resource
+and be notified about changes, instead of having to keep asking.
+
+## Running our controller
+
+It's time to run our controller, thus testing our OIC server and the
+controller itself. On one terminal, run our server:
+
+```sh
+$ sol-fbp-runner oic-and-http-light-server.fbp oic-and-http-light-server.fbp
```

Comment:
On 17-03-2016 19:53, Gustavo Sverzut Barbieri wrote:
> In doc/tutorials/ostro-oic-tutorial/step4/tutorial.md
> <https://github.com/solettaproject/soletta/pull/1645#discussion_r56591146>:
>
>> +
>> +```sh
>> +oic OUT_POWER -> IN _(console:prefix="Server power level is: ")
>> +```
>> +
>> +This way, if another controller changes server 'power' value, we
>> +are informed. That's a nice OIC feature: we can observe the resource
>> +and be notified about changes, instead of having to keep asking.
>> +
>> +## Running our controller
>> +
>> +It's time to run our controller, thus testing our OIC server and the
>> +controller itself. On one terminal, run our server:
>> +
>> +```sh
>> +$ sol-fbp-runner oic-and-http-light-server.fbp oic-and-http-light-server.fbp
>
> again 2 parameters to fbp-runner...

copy/pasta dyslexia...

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1645/files/e0dc10a0a8279b974cc9b71d4902788977f82ce0#r56591146>
>